torch/distributed/_shard/checkpoint/metadata.py

```diff
@@ -16,8 +16,6 @@ class ShardStorageMetadata:
     shard_metadata: ShardMetadata
     # storage key used for this particular Shard
     storage_key: str
-    # Length in bytes for this shard
-    length: int
```

Comment:
Oh did we miss this redundant field earlier because it is already covered by ShardMetadata?

Reply:
It's implicit from shard size + shared tensor element size.